examples/instance/terraform/plugin.go

```diff
@@ -370,8 +370,25 @@ func (p *plugin) Provision(spec instance.Spec) (*instance.ID, error) {
 			}
 		}
 	case "softlayer_virtual_guest":
-		log.Debugln("softlayer_virtual_guest detected, adding hostname to properties: hostname=", name)
-		properties.Value["hostname"] = name
+		// Use the given hostname value as a prefix if it is a non-empty string
+		var hostnamePrefix string
+
+		switch properties.Value["hostname_prefix"].(type) {
```

Comment:
Assuming we can use `@hostname_prefix` without problems, I think this should be fine.  All this extra processing is triggered by the presence of only the `@something` directive in the properties and only when a related `hostname` field is present...  So I think it's pretty safe.